### ISSUE TITLE
Typechecker panics with 'expected Set struct' when a set argument is out of order

### ISSUE THREAD

**Issue by cybergsus** [SerenityOS/jakt#1035]
Given the following code:
```jakt
struct Thing {
    set_of_things: {String}
    another_arg: i32
}

function main() {
    let thing = Thing(another_arg: 3, set_of_things: {})
}
```
the typechecker aborts with `internal error: expected Set struct`. 

Interestingly enough, I could not trigger the error by using an incorrect type inside the set, since this runs correctly:

```jakt
struct Thing {
    set_of_things: {String}
}

function main() {
    let thing = Thing(set_of_things: {1})
}
```
```
Error: Type mismatch: expected ‘String’, but got ‘i64’
----- test.jakt:7:38
 6 | function main() {
 7 |     let thing = Thing(set_of_things: {1})
                                          ^- Type mismatch: expected ‘String’, but got ‘i64’
 8 | }
-----
```

**Comment by robryanx** [comment]
When you include a value it determines the inner type from that.
I think the issue is that it only uses the type hint when the type is a `GenericInstance`
```
if .get_type(type_hint!) is GenericInstance(id, args) and id.equals(set_struct_id) {
```
I guess for any other type it should be okay for `inner_type_id = type_hint!`

**Comment by lanmonster** [comment]
fixed as of 1ea81c6
